flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/batch/sql/DagOptimizationTest.xml

```diff
@@ -273,7 +273,7 @@ Calc(select=[a AS a1, b, c AS c2])(reuse_id=[3])
    :- Exchange(distribution=[hash[a]])
    :  +- Calc(select=[a, b], where=[(a <= 10)])
    :     +- Reused(reference_id=[1])
-   +- Exchange(distribution=[hash[a2]], shuffle_mode=[BATCH])
```

Comment:
What change leads to this difference?

Reply:
Discussed offline. This is caused by switching from all-exchanges-pipelined to all-exchanged-blocking. In all-exchanged-blocking scenario, the exchange is not required to be blocking.

Reply:
However, I think we should keep `DagOptimizationTest` and `DeadlockBreakupTest` to run with all-exchanges-pipelined, because they are testing whether deadlocks are avoided in pipelined shuffle cases.

Reply:
+1